### PR TITLE
Model template

### PR DESCRIPTION
**Pull request by ElieKadoche**
Here I propose a model template.

To get a harmonious solution (code, usage, readability) I propose this structure for all models.

We also need to discuss about #4, which is an important matter before going more deeply into the project.

### CODE REVIEW

File: clean_dataset.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    data_clean = data_clean.get_text()
+
+    # Remove URL http
+    data_clean = re.sub("https?://[A-Za-z0-9./]+", "", data_clean)
+
+    # Remove mention @
+    data_clean = re.sub(r"@[A-Za-z0-9]+", "", data_clean)
+
+    # Remove hastag #, or other informations, keep only letter
+    data_clean = re.sub("[^a-zA-Z]", " ", data_clean)
```
Comment (XanX3601) [inline clean_dataset.py]:
```suggestion
    data_clean = re.sub("#[^a-zA-Z]", " ", data_clean)
```

**Review by XanX3601 — CHANGES_REQUESTED** [review]
Just a minor issue that I have detected otherwise, it seems like a good start !

File: clean_dataset.py (hunk 1 above)
Comment (ElieKadoche) [reply clean_dataset.py]: Corrected in https://github.com/XanX3601/deep_sentiment140/pull/5/commits/f1290653db971dbf3796a5cd9bffd3abf692dfb9.

**Review by XanX3601 — APPROVED** [review]
Good for me !

**Comment by fmdinh** [comment]
Hi,

I just pushed my code for the LSTM model.

A few comments:

- The cleaning file I forgot to delete the UTF-8 BOM in the dataset (\xef \xbf \xbd ...) so I add it to the clean_dataset file.
- 0.2% of tweets are empty after cleaning. So I delete the empty tweet in the preload.py. In this file, I also take 10% of the data set for training (it already took me over an hour to scan with 10% of the data set).

I use Glove vector for embedding.

I used an LSTM model. I get an accuracy of 70% for training but only 54% for tests. I think that is also the structure of the LSTM. In fact, LSTM takes words from the transition sequence, from which I removed the punctuation (if I keep the punctuation, the result will be really worse). It may create a little overfit when the sentences get long due to lack of punctuation.

I think that with the CNN model or your other models will have better results!

Also, I think we should start reporting. May I create a link on the overleaf to share for three of us?

Thank you,

**Pull request merged**